server/API/REST.php

```diff
@@ -524,6 +505,60 @@ function set_lecture_online_status($vars) {
 	}
 }
 
+if(!function_exists('print_courses')){
+	function print_courses($vars){
+		try{
+			$pdo = new PDO('sqlite:../db.sqlite');
+
+			if(isset($_GET["ofLogged"])){
+				$logged_user = get_myself();
+				switch($logged_user["type"]){
+					case USER_TYPE_TEACHER:
+						$stmt = $pdo->prepare("SELECT courses.*, users.firstname AS teacherFirstName, users.lastname AS teacherLastName, users.email AS teacherEmail, users.ID AS teacherId FROM courses, users WHERE courses.teacher_id = users.ID AND courses.teacher_id = :teacherID");
+
+						$stmt->bindValue(":teacherID", $logged_user["userId"]);
+
+						if(!$stmt->execute()){
+							throw new PDOException($stmt->errorInfo()[2]);
+						}
+
+						$courses = $stmt->fetchAll(PDO::FETCH_ASSOC);
+					break;
+					case USER_TYPE_STUDENT:
+						$stmt = $pdo->prepare("SELECT courses.*, users.firstname AS teacherFirstName, users.lastname AS teacherLastName, users.email AS teacherEmail, users.ID AS teacherId FROM courses, users WHERE courses.teacher_id = users.ID AND courses.ID IN ( SELECT course_id FROM course_subscriptions WHERE user_id = :studentID)");
+
+						$stmt->bindValue(":studentID", $logged_user["userId"]);
+
+						if(!$stmt->execute()){
+							throw new PDOException($stmt->errorInfo()[2]);
+						}
+
+						$courses = $stmt->fetchAll(PDO::FETCH_ASSOC);
+					break;
+					default:
+						throw new ErrorException("Your user type does not have assigned courses");
+					break;
+				}
+			}
+			else{
+				$stmt = $pdo->query("SELECT courses.*, users.firstname AS teacherFirstName, users.lastname AS teacherLastName, users.email AS teacherEmail, users.ID AS teacherId FROM courses, users WHERE courses.teacher_id = users.ID");
```

Comment:
Expand courses.* to correctly alias the fields with the names specified in endpoints.md

Reply:
they already does. Check out the endpoint.md

Reply:
teacher_id and teacherID

Reply:
theacherId was just a misspelling of teacherId. Teacher_id is there only to not make more complex the query, it does not harm